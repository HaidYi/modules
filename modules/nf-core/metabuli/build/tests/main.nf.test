nextflow_process {

    name "Test Process METABULI_BUILD"
    script "../main.nf"
    process "METABULI_BUILD"

    tag "modules"
    tag "modules_nfcore"
    tag "metabuli"
    tag "metabuli/build"
    
    
    // setup {
    //     run("GUNZIP") {
    //         script "modules/nf-core/gunzip/main.nf"
    //         process {
    //             """
    //             input[0] = Channel.of([
    //                 [],
    //                 file(
    //                     params.modules_testdata_base_path + "genomics/sarscov2/metagenome/prot.accession2taxid.gz",
    //                     checkIfExists: true
    //                 )
    //             ])
    //             """
    //         }
    //     }
    // }


    test("sarscov2 - sarscov2 DNA and AA") {



        when {
            process {
                """
                
                input[0] = [
                    [ id:'test' ],
                        [
                        file(params.modules_testdata_base_path + "genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
                        ]
                    ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_names.dmp', checkIfExists: true)
                input[2] = file(params.modules_testdata_base_path +'genomics/sarscov2/metagenome/prot_nodes.dmp', checkIfExists: true)
                input[3] = [] // needs a merged.dmp
                input[4] = file(params.modules_testdata_base_path + 'delete_me/metabuli/acc2taxid', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                // { assert snapshot(process.out).match() }
            )
        }

    }

}
