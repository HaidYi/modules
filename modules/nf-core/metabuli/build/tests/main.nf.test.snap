{
    "sarscov2 - sarscov2 DNA": {
        "content": [
            "acc2taxid.map:md5,989a01a1b7aa92b2a3e7204872bf36ec",
            "diffIdx:md5,8ac88d03fabf6917698a02841e410b33",
            "info:md5,f91cf2e0090678b481aa7a5105453232",
            "split:md5,6bfa0743a59482ec0a75f3089b9b523e",
            "taxID_list:md5,367764329430db34be92fd14a7a770ee",
            "db.parameters"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-10T10:05:57.259021745"
    }
}